compiler/rustc_middle/src/thir.rs

```diff
@@ -962,6 +979,9 @@ impl<'tcx> PatRangeBoundary<'tcx> {
     ) -> Option<Ordering> {
         use PatRangeBoundary::*;
         match (self, other) {
+            (PosInfinity, PosInfinity) => return Some(Ordering::Equal),
+            (NegInfinity, NegInfinity) => return Some(Ordering::Equal),
```

Comment:
Should we special-case the other comparisons with infinity? Or do we have too many special cases?

Reply:
We shouldn't, because we want to detect that e.g. `10u8..` and `10u8..=255` have the same upper bound

Reply:
I added a comment